Addition of browser_specific_settings.gecko_android (#26171)
mdn/content: files/en-us/mozilla/add-ons/webextensions/manifest.json/browser_specific_settings/index.md

### Firefox (Gecko) properties

Firefox stores browser-specific settings in these subkeys:

- `gecko` for the desktop version of Firefox
- `gecko_android` for the Android version of Firefox.

The `gecko` subkey supports these properties:

- `id`
  - : The extension ID. For extensions submitted to [addons.mozilla.org](https://addons.mozilla.org/) (AMO), see [Extensions and the Add-on ID](https://extensionworkshop.com/documentation/develop/extensions-and-the-add-on-id/) to determine when to specify the ID. When required by AMO, this property must contain 80 characters or less. For extensions not submitted to AMO, an ID is required if the extension is unsigned (and not loaded using `about:debugging`); otherwise, an ID is optional.
- `strict_min_version`

>   - : Minimum version of Gecko to support. Versions containing a "\*" are not valid in this field. Defaults to "42a1".

Reviewer: I don't think there is a default value, especially not `42a1`.

Author: @willdurand I assume that one of these must apply:
- when not provided there is a default
- is mandatory if you also state `gecko.strict_max_version`
- it takes the value of `gecko.strict_max_version`. 
Can you confirm which one?

Reviewer: > when not provided, there is a default

there is no default AFAICT, it's therefore compatible.

> is mandatory if you also state gecko.strict_max_version

no, min version isn't mandatory even when max version is specified

> when not provided, it takes the value of gecko.strict_max_version.

it does not

> when not provided, (based on your feedback about back at gecko_android) installs on whatever minimum version is supported by AMO

that is different, AMO has its own compatibility range.

Reviewer: the absence of this value means it is compatible

- `strict_max_version`
  - : Maximum version of Gecko to support. If the Firefox version on which the extension is being installed or run is above this version, then the extension is not installed or not run. Defaults to "\*", which disables checking for a maximum version.
- `update_url`
  - : A link to an [extension update manifest](https://extensionworkshop.com/documentation/manage/updating-your-extension/). Note that the link must begin with "https". This key is for managing extension updates yourself (i.e., not through AMO).

The `gecko_android` subkey supports these properties:

- `strict_min_version`

>   - : Minimum version of Gecko to support. Versions containing a "\*" are not valid in this field. Defaults to "42a1".

Reviewer: I don't think the default value is `42a1`. There is no default in Firefox for this property. If this property is omitted, `gecko.strict_min_version` is used instead.

AMO has a default value that might change in the future but we shouldn't document it here anyway.

- `strict_max_version`

>   - : Maximum version of Gecko to support on Android. If the Firefox version on which the extension is being installed or run is above this version, the extension is not installed or not run. Defaults to "\*", which disables checking for a maximum version.

Reviewer: ```suggestion
  - : Maximum version of Gecko to support on Android. If the Firefox version on which the extension is being installed or run is above this version, the extension is not installed or not run. Defaults to the version determined by `gecko.strict_max_version`.
```
